checker/build.gradle

```diff
@@ -119,21 +116,53 @@ task allJavadocJar(type: Jar) {
     from rootProject.tasks.allJavadoc.destinationDir
 }
 
+// Shadowing Test Sources and Dependencies
+import com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar
 
-shadowJar {
-    description 'Creates the "fat" checker.jar in dist/.'
-    destinationDirectory = file("${projectDir}/dist")
-    archiveFileName = "checker.jar"
-    // Without this line, the Maven artifact will have the classifier "all".
+task checkerJar(type: ShadowJar, dependsOn: compileJava, group: 'Build') {
+    description "Builds checker-${project.version}.jar with all dependencies except checker-qual and checker-util."
+    includeEmptyDirs = false
+    archivesBaseName = 'checker'
     archiveClassifier = ''
+
+    from shadowJar.source
+    configurations = shadowJar.configurations
+    // To see what files are incorporated into the shadow jar file:
+    // doLast { println sourceSets.main.runtimeClasspath.asPath }
+    manifest {
+        attributes("Main-Class": "org.checkerframework.framework.util.CheckerMain")
+    }
+    exclude 'org/checkerframework/**/qual/*'
+    exclude 'org/checkerframework/checker/*/util/*'
+    relocators = shadowJar.getRelocators()
+}
+
+jar {
+    dependsOn(checkerJar)
+    // Never build the skinny jar.
```

Comment:
What is the point of this `jar` block, given that it contains `onlyIf {false}`?

Reply:
`jar` is a task automatically added by the Java plugin.  Other tasks automatically added by the Java plugin use this task and assumes it builds the main jar for a project.  So, this block adds a dependency to the task that actually builds the main jar and prevents it from build the standard jar.

Reply:
Could you write some of this in the `build.gradle` file?  I was asking for future me and other maintainers, who won't have this insight.  Thanks.